src/util/logging.cpp

```diff
@@ -51,10 +54,16 @@ static void initLogging(const std::string& name)
 std::shared_ptr<spdlog::logger> getLogger(const std::string& name)
 {
     // Lazy-initialize logger
-    if (loggers.count(name) == 0) {
-        initLogging(name);
+    if (loggers.find(name) == loggers.end()) {
+        faabric::util::FullLock lock(loggerMx);
+        if (loggers.find(name) == loggers.end()) {
+            initLogging(name);
+        }
     }
 
-    return loggers[name];
+    {
+        faabric::util::SharedLock lock(loggerMx);
+        return loggers[name];
+    }
```

Comment:
Same race condition issue mentioned previously. Given the proposed move to using spdlog macros we don't need to worry too much about the performance hit this will have (which will probably be negligible anyway unless called in a very tight loop)